src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTreeNode.scala

```diff
@@ -2,17 +2,27 @@
 
 package freechips.rocketchip.diplomaticobjectmodel.logicaltree
 
-import freechips.rocketchip.diplomacy.ResourceBindingsMap
+import freechips.rocketchip.diplomacy.BindingScope.bindingScopes
+import freechips.rocketchip.diplomacy.{BindingScope, Device, ResourceBindings, ResourceBindingsMap, SimpleDevice}
 import freechips.rocketchip.diplomaticobjectmodel.model.OMComponent
 
 import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
 
-trait LogicalTreeNode {
-  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, children: Seq[OMComponent] = Nil): Seq[OMComponent]
+abstract class LogicalTreeNode(protected val deviceOpt: () => Option[Device]) {
+  def getOMComponents(resourceBindings: ResourceBindings, children: Seq[OMComponent] = Nil): Seq[OMComponent]
+  def getDevice = deviceOpt
```

Comment:
The intent is to remove this down-the road, right?

Reply:
```protected val deviceOpt: () => Option[Device]``` will be removed down the road.